libweb3jsonrpc/Eth.cpp

```diff
@@ -271,6 +271,42 @@ string Eth::eth_sendTransaction(Json::Value const& _json)
 	{
```

Comment:
Please remove this `catch`, it's useless

Reply:
Right you are, I'll remove it.

Reply:
Actually, the catch is required to surface the account / user errors to the user and without it 5 tests in `ClientTests/Personal` fail. I think that it's best to remove the catch though since a cleaner way to surface the errors would be by having `AccountHolder::authenticate` throw on failure and catch / re-throw at the RPC level like I'm doing for the other errors (as we discussed in #3121). As a result, I've updated the client tests with the new generic error message so they should all pass now.

Reply:
I see, this was weird.
Maybe then instead of throwing `JsonRpcException` in the `try` block above we can now throw some other exception (maybe some new `TransactionAuthenticationError`), catch it and rethrow as `JsonRpcException`.

If you plan to refactor `AccountHolder::authenticate` we can defer this.

Reply:
Ah I see the issue you created is exactly about this. Ok then, let's leave like this for now